include/rosbag2_exporter/utils.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef UTILS_HPP
+#define UTILS_HPP
+
+#include <string>
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <vector>
+
+#include "rosbag2_exporter/handlers/base_handler.hpp"
+#include "builtin_interfaces/msg/time.hpp"
+
+namespace rosbag2_exporter {
+
+std::string get_cam_name(const std::string& path) {
+    const std::string prefix = "/sensor/camera/";
+    size_t start = path.find(prefix);
+    if (start == std::string::npos) {
+        return ""; // Prefix not found
+    }
+    
+    // Move past "/sensor/camera/"
+    start += prefix.length(); 
+    
+    size_t end = path.find('/', start);
+    return path.substr(start, end - start);
+}
+
+// Helper function to convert builtin_interfaces::msg::Time to nanoseconds
+int64_t toNanoseconds(const builtin_interfaces::msg::Time& timestamp) {
+    return static_cast<int64_t>(timestamp.sec) * 1'000'000'000 + timestamp.nanosec;
```

Comment:
Same thing as with the python scripts (Although it is nicer to see the number split with ' apostrophes).
For clarity and to avoid mistakes I would write `std::pow(10, 9)` after `#include<cmath>`